pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java

```diff
@@ -1219,7 +1219,43 @@ public void resetCursorOnPosition(
             MessageIdImpl messageId) {
         try {
             validateTopicName(tenant, namespace, encodedTopic);
-            internalResetCursorOnPosition(asyncResponse, decode(encodedSubName), authoritative, messageId);
+            internalResetCursorOnPosition(asyncResponse, decode(encodedSubName), authoritative, messageId, false);
+        } catch (Exception e) {
+            resumeAsyncResponseExceptionally(asyncResponse, e);
+        }
+    }
+    
+    @POST
+    @Path("/{tenant}/{namespace}/{topic}/subscription/{subName}/resetCursorExclusive")
+    @ApiOperation(value = "Reset subscription to message position closest to given position, " +
+            "and start consume messages from the next position of the reset position.", notes = "It fence cursor and disconnects all active consumers before reseting cursor.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 307, message = "Current broker doesn't serve the namespace of this topic"),
+            @ApiResponse(code = 401, message = "Don't have permission to administrate resources on this tenant or" +
+                    "subscriber is not authorized to access this operation"),
+            @ApiResponse(code = 403, message = "Don't have admin permission"),
+            @ApiResponse(code = 404, message = "Topic/Subscription does not exist"),
+            @ApiResponse(code = 405, message = "Not supported for partitioned topics"),
+            @ApiResponse(code = 412, message = "Unable to find position for position specified"),
+            @ApiResponse(code = 500, message = "Internal server error"),
+            @ApiResponse(code = 503, message = "Failed to validate global cluster configuration") })
+    public void resetCursorOnPositionExclusive(
```

Comment:
I think we'd better do not introduce a new endpoint since we also should support reset to a `batchIndex`. So it's better to introduce a new construct for the cursor reset request such as `ResetCursorData` and we and add the `isExcluded` into the new construct, it looks like

```
class ResetCursorData {
    ledgerId,
    entryId,
    isExcluded
}
```

Since we are using the json encoder and decode, I think this will not introduce compatibility issues.